biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -338,17 +339,25 @@ private boolean isMulti(String name) {
 	}
 
 	private Map<Archive, Resource> parseMulti(Archive archive, File multi) throws Exception {
+		Set<String> multis = new HashSet<String>();
+		File localArchives = repo.toLocalFile(archive)
+			.getParentFile();
 		Map<Archive, Resource> result = new HashMap<>();
 		try (Jar jar = new Jar(multi)) {
-			int n = 1000;
 			for (Entry<String, aQute.bnd.osgi.Resource> entry : jar.getResources()
 				.entrySet()) {
 				String path = entry.getKey()
 					.toLowerCase();
 				if (path.endsWith(".jar")) {
+					String name = Strings.lastPathSegment(path)[1];
```

Comment:
`lastPathSegment` can return `null`. So you need to guard against NPE here.

Reply:
But he checks for .jar at the end? So no null check needed as far as I can see?

Reply:
done

Reply:
@pkriens I believe BJ is right. if the jar is at the root of the zip, I can see a path like "somejar.jar", in which case `lastPathSegment` would return null.